docs/maintainers-guide.md

```diff
@@ -0,0 +1,174 @@
+```
+git clone git@github.com:hashgraph/hedera-services.git
+```
+
+From IntelliJ, choose `File -> Open` the _hedera-services/_ directory you just cloned.
+Make sure you are using JDK17 as the project SDK in IntelliJ.
+
+<p>
+    <img src="assets/jdk-17.png"/>
+</p>
+
+## Gradle
+Once the repository is opened in IntelliJ, to build the project from Intellij, open the Gradle tool window and
+run `Tasks/build/assemble` to on the root project. If you are using command line use `./gradlew assemble`.
+
+Start using either the Gradle command line
+`(./gradlew spotlessApply)`  or set up the [Google Java Format IntelliJ Plugin](https://github.com/google/google-java-format#intellij-android-studio-and-other-jetbrains-ides)
+in IntelliJ to format your code to follow Google Code Style.
+
+## GPG set up
+Every commit being pushed to the repository should be verified and [signed off](#dco-sign-off). So it is important to set up GPG keys before
+contributing to the repository.
+
+Use the following tutorials to set up a GPG key.
+
+**Be sure to enable Vigilant Mode and adding GPG key in GitHub**.
+
+- **Github**
+  - [Github - Generating a new GPG key](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key)
+  - [Github - Adding a GPG key to your Github account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account)
+  - [Github - Configuring your Git CLI for GPG commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key)
+  - [Github - Signing Commits with the Git CLI](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+  - [Github - Vigilant Mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits) ‼️
+- **IntelliJ IDEA (if you use IntelliJ to interact with git)**
+  - [IntelliJ - Signing Commits with GPG Keys](https://www.jetbrains.com/help/idea/set-up-GPG-commit-signing.html)
+  - [IntelliJ Official YouTube - GPG Commit Signing (10:59)](https://youtu.be/RBhz-8fZN9A?t=659)
+
+## Development Model
+The Git Source Code Management(SCM) system treats branches as one of the fundamental constructs for supporting 
+concurrent development and assisting with merge conflict resolution. Due to the fundamental nature and flexibility 
+offered by Git branches, it is critical to have a consistent branching workflow in order to reduce merge conflicts and 
+support automation.
+
+Git commit tags play a critical role in Release Management strategies and automated releases. Tags are affected by
+branches and branch related operations; therefore, it is critical that the branching workflow incorporates a consistent 
+tagging strategy.
+
+[GitFlow branching model](https://nvie.com/posts/a-successful-git-branching-model/) is elected pattern for the 
```

Comment:
[GitFlow branching model](https://nvie.com/posts/a-successful-git-branching-model/) is the team's chose pattern